src/nisarqa/utils/utils.py

```diff
@@ -133,6 +133,50 @@ def raise_(exc):
     raise exc
 
 
+def _prep_data_for_hdf5_using_nisar_conventions(
+    data: ArrayLike | str | bool,
+) -> ArrayLike | np.bytes_:
+    """Prep data for writing to the STATS.h5 file, per NISAR conventions."""
+
+    # If a string or a list of strings, convert to fixed-length byte strings
+
+    # If `data` is an e.g. numpy array with a numeric dtype,
+    # do not alter it.
+    if isinstance(data, str) or (
+        isinstance(data, Sequence) and all(isinstance(s, str) for s in data)
+    ):
```

Comment:
The order of these comments looks weird.

I think the first comment should be attached to the `if` statement here, and the second comment should perhaps go to the end of the function, before we `return data`.